### PR TITLE
Updates repated to supported Unity/CppUTest frameworks

### PR DESCRIPTION
**Pull request by PrzemekWirkus**

### CODE REVIEW

**Comment by 0xc0170** [comment]
Shall we add utest reference? It provides greentea handlers so it's greentea friendly :-)

**Comment by PrzemekWirkus** [comment]
Yes, we will provide Unity (utest?) reference, how about we talk about this.
We can add it here: https://github.com/ARMmbed/greentea/issues/39 to solve this issue.

**Comment by 0xc0170** [comment]
> Yes, we will provide Unity (utest?) reference, how about we talk about this.

Yes, cpputest is not public anyway, so this one can stay open if we consider cpputest to be useful with greentea.  I would like to see here utest, bringing @niklas-arm here.  
Not certain how descriptive we should be, but at least a pointer to utest module and its documentation?

File: docs/QUICKSTART.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ This document is a simple cookbook introducing the testing facilities available
 
 This document provides examples for all three methods of testing.
 
-* You can also create [unit tests](http://en.wikipedia.org/wiki/Unit_testing) with the [mbed compatible CppUTest library](https://github.com/ARMmbed/mbed-cpputest-private). Information about the CppUTest library can be found [here](https://cpputest.github.io/manual.html).
+* You can also create [unit tests](http://en.wikipedia.org/wiki/Unit_testing) with the mbed compatible CppUTest library. Information about the CppUTest library can be found [here](https://cpputest.github.io/manual.html) or 
+* use [Unity framework](https://github.com/ThrowTheSwitch/Unity) supported by mbed: [Unit](https://github.com/ARMmbed/unity) framework.
```
Comment (niklas-arm) [inline docs/QUICKSTART.md]: Unit => Unit**y**

**Comment by niklas-arm** [comment]
The `utest` Readme already contains an explicit example with greentea, so I would only point people there via a link and not mirror parts of the Readme here.

Our Unity fork only works with `utest`, as it has specific changes made so that the reporting works!

If people want to use Unity as a stand alone framework, they must use the official repo.

**Comment by PrzemekWirkus** [comment]
I've mentioned utest and added link to example in documentation for ```greentea```.

File: docs/QUICKSTART.md [hunk 2]
```diff
@@ -8,7 +8,9 @@ This document is a simple cookbook introducing the testing facilities available
 
 This document provides examples for all three methods of testing.
 
-* You can also create [unit tests](http://en.wikipedia.org/wiki/Unit_testing) with the [mbed compatible CppUTest library](https://github.com/ARMmbed/mbed-cpputest-private). Information about the CppUTest library can be found [here](https://cpputest.github.io/manual.html).
+* Use [Unity framework](https://github.com/ThrowTheSwitch/Unity) supported by mbed: [Unity](https://github.com/ARMmbed/unity) framework.
+  * Please have a look at our Unity integration under lean test hardness called [utest: A Simple C++ Test Harness](https://github.com/ARMmbed/utest). Utest documentation provides [examples how to use it with Greentea](https://github.com/ARMmbed/utest#example).
```
Comment (niklas-arm) [inline docs/QUICKSTART.md]: "lean test hardness"?

**Pull request merged**